### PR TITLE
gh-83365: Fix incorrect return value in `msvcrt_get_osfhandle_impl`

### PR DESCRIPTION
**Pull request by gousaiyang**
As per #18580, `msvcrt_get_osfhandle_impl` should also return -1 instead of `NULL` when `PySys_Audit` fails.

https://bugs.python.org/issue39184

* Issue: gh-83365

### CODE REVIEW

**Comment by gousaiyang** [comment]
Hi @zooba, could you review this PR if you have time? This PR seems necessary to correct a mistake I made previously.

**Review by nanjekyejoannah — COMMENTED** [review]
Dont we need any tests for this change?

**Review by zooba — APPROVED** [review]

**Comment by zooba** [comment]
If we had failure tests for all APIs with audit hooks, then I'd insist upon it. But we don't, and this only would've been caught by having broad testing (100+ new tests).

It's worth creating an issue for adding that set of tests. They ought to be fairly quick as well (though some will require e.g. making a real network connection, and most will require starting a new Python process for each API), but I think we can take this fix without adding all that.

**Review by kumaraditya303 — CHANGES_REQUESTED** [review]
The CLA needs to be signed.

File: PC/msvcrtmodule.c [hunk 1]
```diff
@@ -209,7 +209,7 @@ msvcrt_get_osfhandle_impl(PyObject *module, int fd)
 /*[clinic end generated code: output=aca01dfe24637374 input=5fcfde9b17136aa2]*/
 {
     if (PySys_Audit("msvcrt.get_osfhandle", "(i)", fd) < 0) {
-        return NULL;
+        return -1;
```
Comment (zooba) [inline PC/msvcrtmodule.c:212]:
```suggestion
        return (void*)-1;
```

**Comment by zooba** [comment]
Nothing heard from the OP, but also it doesn't seem to be necessary. This function call checks for any error, without looking at the return value.

**Pull request closed without merging**